### PR TITLE
Add Live Reload overlay to the client

### LINKED ISSUE #7790: Client-side UI for live reload
- Indicator shown on top of application with "Disable" button, shown only in development mode. Ok that the indicator is always overlaid on the application.
- "Disable" button should disable automatic reload and store the setting in local storage as a Boolean. It can be shown as a link in the indicator. Currently clearing the cache will be the only way to re-enable it for now.
- Use the below UI mockup as a hint for how it should look. Do not polish the CSS too much at this point yet.
- Open websocket to server immediately when client is loading in dev mode. When the client gets the handshake message `{ "command": "hello" }` it will show the message "Live reload enabled" (see UI mockup below) for 5 seconds.

screenshot_2020-03-05_at_19 08 11

### LINKED ISSUE #7793: Spring Boot Dev Tools integration with Flow Live Reload client
[Client-side UI for live reload](https://github.com/vaadin/flow/issues/7790) should be able to connect to Spring Boot Dev Tools to reload the page when Dev Tools restarts the app on the server. 
See https://github.com/vaadin/flow/issues/7740#issuecomment-594895771 for implementation tips.

### PR DESCRIPTION
**Pull request by mehdi-vaadin**
Fixes #7790 
Fixes #7793

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/flow/7829)

### CODE REVIEW

File: flow-client/src/main/resources/com/vaadin/ClientEngine.gwt.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <!DOCTYPE module PUBLIC "-//Google Inc.//DTD Google Web Toolkit 2.8.0//EN" "http://gwtproject.org/doctype/2.8.0/gwt-module.dtd">
 <module rename-to="client">
 	<inherits name="com.vaadin.ClientEngineXSI" />
+    <inherits name="com.google.gwt.storage.Storage"/>
```
Comment (bogdanudrescu) [inline flow-client/src/main/resources/com/vaadin/ClientEngine.gwt.xml]: This seems misalign with indentation.

File: flow-client/src/main/resources/com/vaadin/ClientEngineXSI.gwt.xml [hunk 2]
```diff
@@ -14,8 +14,9 @@
 
 	<inherits name="com.google.gwt.useragent.UserAgent" />
 	<inherits name="com.google.gwt.precompress.Precompress" />
+    <inherits name="com.google.gwt.storage.Storage"/>
```
Comment (bogdanudrescu) [inline flow-client/src/main/resources/com/vaadin/ClientEngineXSI.gwt.xml]: This seems misalign with indentation.

File: flow-client/src/main/resources/com/vaadin/ClientEngineXSI.gwt.xml [hunk 3]
```diff
@@ -14,8 +14,9 @@
...
 
-	<source path="client" />
+    <source path="client" />
```
Comment (bogdanudrescu) [inline flow-client/src/main/resources/com/vaadin/ClientEngineXSI.gwt.xml]: This seems misalign with indentation.

File: flow-client/src/main/resources/com/vaadin/ClientEngine.gwt.xml (hunk 1 above)
Comment (bogdanudrescu) [reply flow-client/src/main/resources/com/vaadin/ClientEngine.gwt.xml]: Or maybe the other lines are not properly indented (same for the others)

**Comment by mehdi-vaadin** [comment]
This PR might need some fixes via other PR(s) to fix test failures here. It's still WIP.

**Pull request merged**